webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Storages/StorageConfig/AdditionalConstraints/ConstraintForm/OccurrencesFE/OccurrencesTableDialog/index.tsx

```diff
@@ -0,0 +1,227 @@
+  onEdit: (occurrences: ConstraintValues["occurrences"]) => void;
+}
+
+function OccurrencesTableDialog({ open, onClose, onEdit, control, occurrences }: Props) {
+  const { t } = useTranslation();
+  const [columnCount, setColumnCount] = useState(occurrences.length);
+  const [offset, setOffset] = useState(0);
+  const dataGridApiRef = useRef<DataGridFormApi<OccurrencesData>>(null);
+  const [selectedHours, setSelectedHours] = useState<number[]>(R.range(1, HOURS_IN.WEEK + 1));
+  const closeAction = useConfirm();
+
+  const name = useWatch({
+    name: "name",
+    control,
+  });
+
+  const columns = useMemo<DataGridFormProps["columns"]>(() => {
+    return Array.from({ length: columnCount }, (_, index) => ({
+      id: index.toString(),
+      title: `${t("global.occurrence")} ${index + 1}`,
+    }));
+  }, [columnCount, t]);
+
+  const defaultData = useMemo<OccurrencesData>(() => occurrencesToGridData(occurrences), []);
+
+  ////////////////////////////////////////////////////////////////
+  // Event Handlers
+  ////////////////////////////////////////////////////////////////
+
+  const handleOffsetChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = Number(event.target.value);
+    setOffset(R.clamp(0, HOURS_IN.WEEK - 1, value));
+  };
+
+  const handleResize = (value: number) => {
+    if (dataGridApiRef.current) {
+      const { data, setData } = dataGridApiRef.current;
+
+      setColumnCount(value);
+
+      const resize = resizeRow(value);
+      setData(R.map(resize, data));
+    }
+  };
+
+  const handleSubmit = (
+    { values }: SubmitHandlerPlus<OccurrencesData>,
+    event: React.FormEvent<HTMLFormElement>,
+  ) => {
+    // Prevent to submit the constraint form
+    event.stopPropagation();
+
+    const occurrences = gridDataToOccurrences(values);
+
+    const validation = validateOccurrences(occurrences);
+    if (validation !== true) {
+      throw new Error(validation);
+    }
+
+    onEdit(occurrences);
+
+    onClose();
+  };
+
+  const handleClose = async () => {
+    if (dataGridApiRef.current?.formState.isDirty) {
+      const isConfirmed = await closeAction.showConfirm();
+
+      if (!isConfirmed) {
+        return;
+      }
+    }
+
+    onClose();
+  };
+
+  const handleUpdateHours = (updateFn: RA.Pred) => () => {
+    if (dataGridApiRef.current) {
+      const { data, setData } = dataGridApiRef.current;
+
+      const selectedHoursByOccurrence = getSelectedHoursByOccurrence({
+        hours: selectedHours,
+        occurrences: columnCount,
+        offset,
+      });
+
+      const newData = R.mapObjIndexed((rowData, key) => {
+        const hour = Number(key);
+
+        return R.mapObjIndexed((active, key) => {
+          const occurrenceIndex = Number(key);
+          return selectedHoursByOccurrence[occurrenceIndex]?.includes(hour)
+            ? updateFn(active)
+            : active;
+        }, rowData);
+      }, data);
+
+      setData(newData);
+    }
+  };
```

Comment:
Minor: Entered values won’t take effect until the field loses focus (after inputing values we have to click outside the field first, then click on "activate")


![Image](https://github.com/user-attachments/assets/90a3b477-a574-4474-9268-8e2ad44fc356)

Reply:
I cannot reproduce